plugins/image-prioritizer/class-image-prioritizer-img-tag-visitor.php

```diff
@@ -44,6 +44,38 @@ public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 		return false;
 	}
 
+	/**
+	 * Computes responsive sizes for the current element based on its boundingClientRect width captured in URL Metrics.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param OD_Tag_Visitor_Context $context Context.
+	 * @return non-empty-string[] Computed sizes.
+	 */
+	private function compute_sizes( OD_Tag_Visitor_Context $context ): array {
+		$sizes = array();
+
+		$xpath = $context->processor->get_xpath();
+		foreach ( $context->url_metric_group_collection as $group ) {
+			$element_max_width = 0;
+			foreach ( $group->get_xpath_elements_map()[ $xpath ] ?? array() as $element ) {
+				$element_max_width = max( $element_max_width, $element->get_bounding_client_rect()['width'] );
+			}
```

Comment:
What does this do? Is this to ensure that the width used is the maximum width the image had between all samples in the specific viewport group?

Reply:
Yes, exactly.

Reply:
Added comments to make this more explicit in 6045b18ebb9d558870353e882c2a7938bf134bfa